### ISSUE TITLE
Cluster with MSI - adal: Failed to unmarshal the service principal token during refresh. 

### ISSUE THREAD

**Issue by lodejard** [Azure/aad-pod-identity#303]
**Describe the bug**
Configured AzureIdentityBinding for pod `external-dns` with the new support for user-assigned-identity in the cluster's azure.json file.

99% working - seems to be one last json unmarshalling problem getting the token back to the caller?

**Steps To Reproduce**
Configure external-dns to update cluster DnsZone records using the cluster's managed identity.
Values used for helm deployment:
```
  # https://github.com/helm/charts/tree/master/stable/external-dns
  # https://github.com/helm/charts/tree/master/stable/external-dns/values.yaml
  # https://github.com/kubernetes-incubator/external-dns

  provider: azure

  rbac:
    create: true

  nodeSelector:
    beta.kubernetes.io/os: linux

  securityContext: 
    runAsUser: 0   # container runs as root in order to use cluster's MSI identity
```
After installing aad-pod-identity to cluster - create AzureIdentity and AzureIdentityBinding to associate a user-assigned identity with the pod.

The MIC correctly adds the managed identity to linux agent vm, and the NMI appears to understand the association between the calling pod and the allowed identity, but there is an error being raised at some point during the return path.

An error states apparently valid token response can't be unmarshaled because the expires_in field data time is string... Example error from NMI and the ext-dns logs is below.

**Expected behavior**
The external-dns pod will begin using the MSI token provided by NMI.

**AAD Pod Identity version**
Current commit, `68cc55c7bfee1d5ddc40232c9691900eb9abd5ca`

**Kubernetes version**
1.15.0

**Additional context**
Log lines what happen each time the external-dns pod tried to get an access token. It's unclear which parts of the message are from error in the nmi or the ext-dns process.

```
[nmi-jdpwh] time="2019-07-24T01:38:15Z" level=info msg="matched identityType:0 clientid:7efdc--redacted--cb70 resource:https://management.core.windows.net/" req.method=GET req.path=/metadata/identity/oauth2/token req.remote=10.240.0.6
[external-dns-788c474cc9-zkd5g] time="2019-07-24T01:38:15Z" level=error msg="azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/cd0f--redacted--1353/resourceGroups/redacted-rg/providers/Microsoft.Network/dnsZones?api-version=2016-04-01: StatusCode=0 -- Original Error: adal: Failed to unmarshal the service principal token during refresh. Error = 'json: cannot unmarshal number into Go struct field Token.expires_in of type string' JSON = '{\"access_token\":\"eyJ0eXAi--redacted--gb4KlFL6w\",\"refresh_token\":\"\",\"expires_in\":28800,\"expires_on\":1563958345,\"not_before\":1563929245,\"resource\":\"https://management.core.windows.net/\",\"token_type\":\"Bearer\"}'"
```

**Comment by kkmsft** [comment]
Please refer to: https://github.com/Azure/aad-pod-identity/issues/275
Kindly update the azure ada sdk and let us know if the issue still persists.

**Comment by lodejard** [comment]
Yes, that looks like exactly the problem. Thank you. I'll see what I can do to verify the fix and will let you know the results.

That said - because this is a third party docker image we do not build or host I'm not sure if that workaround will scale... But first I'll verify it does fix the issue and will get back to you.

**Comment by lodejard** [comment]
Confirmed - you're absolutely correct that was the underlying problem.

I verified this by making a small change to how NMI re-marshals the token response - to return the `json.Number` fields in the same quoted format as the underlying metadata endpoint. The `external-dns` pod (running the unmodified public docker image) is now working 100% correctly given a user assigned identity from `aad-pod-identity`.

Also submitted a PR #306 if you would like to include this change to increase adal client compatibility. Thanks again!
